easybuild/tools/run.py

```diff
@@ -558,13 +561,30 @@ def to_cmd_str(cmd):
         if fail_on_error:
             raise_run_shell_cmd_error(res)
 
+    # check that we still are in a sane environment after command execution
+    # safeguard against commands that deleted the work dir or missbehaving filesystems
+    try:
+        os.getcwd()
+    except FileNotFoundError:
+        _log.warning(
+            f"Shell command `{cmd_str}` completed successfully but left the system in a unknown working directory. "
+            f"Changing back to initial working directory: {initial_work_dir}"
+        )
+        try:
+            os.chdir(initial_work_dir)
+        except OSError as err:
+            raise EasyBuildError(f"Failed to return to {initial_work_dir} after executing command `{cmd_str}`: {err}")
+        else:
+            if not os.path.isdir(work_dir):
+                work_dir = initial_work_dir
```

Comment:
@lexming What's the reasoning behind this?

Won't this lead to incorrect/misleading information being passed to the `post_run_shell_cmd` hook?

It's not because the working directory in which the command was executed no longer exists that we should "lie" to the post hook?

```suggestion
```

Reply:
`work_dir` is passed to the hooks to signal in which directory the command was run, it's up to the hook what it does with this (could be custom logging, etc.).

If the hook choses to try and change to this directory, it should take into account that this directory may no longer exist. The current working directory when the hook is run will already be an existing directory, so that part is taken care of.

I prefer passing the "correct" working directory down to the hook, rather than lying to the hook (it can detect by itself that the current working directory is different from the `work_dir` value, if it wants to).

Reply:
ok, I get your point. Fixed in https://github.com/easybuilders/easybuild-framework/pull/4390/commits/38ca67d66427b967865304a6f9876b0e0d58df5b